### ISSUE TITLE
Warning: Over 50 classes were generated for component...

### ISSUE THREAD

**Issue by davidalekna** [styled-system/styled-system#587]
Hello. Thanks for this awesome lib! After updating to latest styled-components@beta I have an issue reusing my Flex component. 

```
export const Flex = styled.div`
  display: flex;
  /*  */
  ${flex};
  ${alignItems};
  ${justifyContent};
  ${flexDirection};
  /*  */
  ${space};
  ${layout};
  ${width};
  ${fontSize};
  ${color};
`;
```

I use this Flex container everywhere in my code with styled-system attributes. After updating to styled-components latest beta version `5.0.0-beta.5-ej2` I get this warning message

`
Over 50 classes were generated for component globals__Flex. This happens when some of the props you use for styling have many potential values and we need to make a new CSS class for each variant. Over time the stylesheet will grow and slow down your app.
For these particular CSS rules with high dynamicity, consider using the attrs() method together with a style object.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))`width: 100%;`

  <Component />
`

**Comment by ericberens** [comment]
You should be able to [`compose`](https://styled-system.com/api#compose) your interpolation functions together to resolve this type of error.

**Comment by jxnblk** [comment]
Yes, with v5 the category-based functions have replaced the single prop ones, so `compose` should dedupe style functions that have overlay, but with your example, it would be better to use `layout` (which includes `width` and other props) and not use the shim for v4's `width` style function

**Comment by davidalekna** [comment]
So I've used all of the v5 ones 

```
export const Flex = styled.div`
  display: flex;
  ${flexbox};
  ${space};
  ${layout};
  ${typography};
  ${color};
`;
```

but still getting same error :/
